src/webui/www/private/scripts/client.js

```diff
@@ -767,9 +767,9 @@ window.addEventListener("DOMContentLoaded", () => {
                     torrentsFilterInputTimer = -1;
 
                     let torrentsTableSelectedRows;
-                    let update_categories = false;
-                    let updateTags = false;
-                    let updateTrackers = false;
+                    let update_categories = (category_list.size === 0);
+                    let updateTags = (tagList.size === 0);
+                    let updateTrackers = (trackerList.size === 0);
```

Comment:
Shouldn't them be initialized after `if (full_update) {}`?

Reply:
Good catch, current solution works but it is not ideal. I think it will be enough to just set all of them to true in the full_update block - data structures are cleared there so no need to check their sizes. What do you think?

Reply:
>What do you think?

👍